### ISSUE TITLE
Bad substitution not reported

### ISSUE THREAD

**Issue by acini** [koalaman/shellcheck#48]
``` bash
#!/bin/bash
var=ab
echo "${"$(echo "${var//a/a1}")"//b/b1}"
exit 0
```

this erronous script reports bad substitution when run in bash shell but reports OK with shellcheck.

**Comment by mklement0** [comment]
:+1: 

Another example of a bad substitution that is not recognized:

	#!/bin/bash
	
	echo ${$p:0:13}

**Comment by kurahaupo** [comment]
FIXED

**Comment by brother** [comment]
[SC2298](https://www.shellcheck.net/wiki/SC2298) (error): ${$x} is invalid. For expansion, use ${x}. For indirection, use arrays, ${!x} or (for sh) eval.
